From 4a99f6dc8f518073b35ba52c503a22de059a89cf Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Mon, 6 Mar 2023 17:01:16 +0000
Subject: [PATCH] CMakeLists.txt: fix build race-condition on generated header
 files

:Release Notes:
Fixes:
http://gecko.lge.com:8000/Errors/Details/547372

:Detailed Nots:
CMake Error:
  Running
   'ninja' '-C' 'TOPDIR/BUILD/work/qemux86_64-webos-linux/libpbnjson/2.15.0-7-r9/build' '-t' 'recompact'
  failed with:
   ninja: error: build.ninja:1005: multiple rules generate src/pbnjson_c/validation/schema_keywords.h [-w dupbuild=err]

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRP-10636] Fix libpbnjson to build with ninja
[WRP-10791] CCC: Various build fixes

Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/libpbnjson/+/346954 CMakeLists.txt: fix build race-condition on generated header files]

Change-Id: I573017789ddddc1eaae41c102700607ad4bdf333
---
 src/pbnjson_c/validation/CMakeLists.txt | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/pbnjson_c/validation/CMakeLists.txt b/src/pbnjson_c/validation/CMakeLists.txt
index 50e1237..b592307 100644
--- a/src/pbnjson_c/validation/CMakeLists.txt
+++ b/src/pbnjson_c/validation/CMakeLists.txt
@@ -50,12 +50,8 @@ include_directories(
 	${API_HEADERS} ${API_HEADERS}/pbnjson ${API_HEADERS}/pbnjson/c
 	)
 
-add_custom_target(
-	schema_keywords echo "Creating jscon_schema_keywords.h"
-	)
-
 function(gperf_generate source target)
-	add_custom_target(${target} echo "Creating ${target}")
+	set_source_files_properties(${target} GENERATED)
 	add_custom_command(
 		SOURCE ${source}
 		TARGET ${target}
